### PR TITLE
Support hrids in RefreshMetadataAction and refactor.

### LINKED ISSUE #4411: Support folio instance HRID in RefreshMetadataAction
RefreshMetadataAction should support folio catalog links / folio instance HRIDs.

In addition, consider the following refactors:
* Rename RefreshMetadataAction (RefreshDescriptionFromCatalog?).
* Get rid of ModsService.
* Drop separate action of getting identifiers from performing refresh.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #4411

## Why was this change made? 🤔
Support HRIDs. Simplification.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit, QA

### CODE REVIEW

File: app/controllers/metadata_refresh_controller.rb [hunk 1]
```diff
@@ -5,29 +5,19 @@ class MetadataRefreshController < ApplicationController
...
-    json_api_error(status: :bad_request, title: 'Catkey not found in Symphony', message: e.message)
+    json_api_error(status: :bad_request, title: 'Not found in catalog', message: e.message)
   rescue Catalog::MarcService::MarcServiceError => e
     Honeybadger.notify(e)
     json_api_error(status: :internal_server_error, message: e.message)
   end
-
-  private
-
-  def identifiers
```
Comment (justinlittman) [inline app/controllers/metadata_refresh_controller.rb:25]: Got rid of the getting identifiers as a separate step in invoking RefreshMetadataAction. This is now combined in the `.run` invocation.

File: app/services/mods_service.rb [hunk 2]
```diff
@@ -1,60 +0,0 @@
-# frozen_string_literal: true
```
Comment (justinlittman) [inline app/services/mods_service.rb:1]: I couldn't figure out the purpose of this class. The handling of identifiers was excessively complicated and the caching seemed unnecessary.

File: app/services/refresh_description_from_catalog.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  def identifiers
+    {
+      catkey:,
+      folio_instance_hrid:,
+      barcode:
+    }.compact
+  end
+
+  def catkey
+    return nil if Settings.enabled_features.read_folio
```
Comment (justinlittman) [inline app/services/refresh_description_from_catalog.rb:56]: Notice that catkey / HRID are controlled by feature flag.

File: app/services/catalog/marc_service.rb [hunk 4]
```diff
@@ -26,16 +26,15 @@ def initialize(catkey: nil, barcode: nil, folio_instance_hrid: nil)
...
 
     # @return [Nokogiri::XML::Document] MODS XML
     # @raise CatalogResponseError
     # @raise CatalogRecordNotFoundError
     def mods_ng
-      marc_xml = marcxml_ng
-      begin
-        marc_to_mods_xslt.transform(marc_xml)
+      @mods_ng ||= begin
+        marc_to_mods_xslt.transform(marcxml_ng)
```
Comment (ndushay) [inline app/services/catalog/marc_service.rb:37]: the dance with altering fields 001 and 003 (or something) from Folio before this xform is a diff ticket, right?

File: app/services/refresh_description_from_catalog.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  def initialize(cocina_object:, druid:, use_barcode:)
+    @cocina_object = cocina_object
+    @druid = druid
+    @use_barcode = use_barcode
+  end
+
+  # @return [Dry::Monads::Results] Returns Failure if it didn't retrieve anything, otherwise Success (Result with description_props and mods)
+  # @raises Catalog::MarcService::MarcServiceError
+  def run
+    # Admin policies don't have identification.
```
Comment (ndushay) [inline app/services/refresh_description_from_catalog.rb:25]: comment reads a little funny as is;  maybe subtitute "Admin policies don't populate cocina identification" ?

File: app/services/refresh_description_from_catalog.rb [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    return Failure() if cocina_object.admin_policy?
+    # No identifiers to refresh from.
+    return Failure() unless identifiers.any?
```
Comment (ndushay) [inline app/services/refresh_description_from_catalog.rb:28]: I'm sure I don't grok this - but is returning Failure() giving enough context for error spelunking?  I dont even understand what "Failure()" means here

File: app/services/mods_service.rb (hunk 2 above)
Comment (ndushay) [reply app/services/mods_service.rb:1]: i found it annoying too when I was tracing the code.  Thanks for killing it.

File: app/services/create_object_service.rb [hunk 7]
```diff
@@ -76,10 +76,7 @@ def merge_access_for(cocina_object)
   def sync_from_catalog(cocina_request_object, druid)
     return cocina_request_object if cocina_request_object.admin_policy?
 
-    catkeys = RefreshMetadataAction.identifiers(cocina_object: cocina_request_object)
-    return cocina_request_object if catkeys.blank?
-
-    result = RefreshMetadataAction.run(identifiers: catkeys, cocina_object: cocina_request_object, druid:)
```
Comment (ndushay) [inline app/services/create_object_service.rb:82]: nice

File: app/services/refresh_description_from_catalog.rb [hunk 8]
```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# Use fetch descriptive metadata from catalog and update description.
```
Comment (ndushay) [inline app/services/refresh_description_from_catalog.rb:3]: "Use fetch" ?

**Review by ndushay — APPROVED** [review]
approving with a few comment wording suggestions

File: app/services/catalog/marc_service.rb (hunk 4 above)
Comment (justinlittman) [reply app/services/catalog/marc_service.rb:37]: That is handled by SymphonyReader and FolioReader.

File: app/services/refresh_description_from_catalog.rb (hunk 8 above)
Comment (justinlittman) [reply app/services/refresh_description_from_catalog.rb:3]: Fixed.

File: app/services/refresh_description_from_catalog.rb (hunk 5 above)
Comment (justinlittman) [reply app/services/refresh_description_from_catalog.rb:25]: They literally don't have identification in the model. That is, there is no `AdminPolicy.identification` (where there is a `Dro.identification`, which may or may not be populated).

Open to other wording.

File: app/services/refresh_description_from_catalog.rb (hunk 6 above)
Comment (justinlittman) [reply app/services/refresh_description_from_catalog.rb:28]: Failure in this context means nothing changed. This is not an error, as it is perfectly expected that not all objects will have desciption refreshed.

I'll update the comment to make clearer.

**Pull request merged**